src_c/mixer.c

```diff
@@ -760,6 +760,22 @@ snd_get_raw(PyObject *self, PyObject *_null)
                                      (Py_ssize_t)chunk->alen);
 }
 
+static PyObject *
+snd_copy(PyObject *self, PyObject *_null)
+{
+    Mix_Chunk *old_chunk = pgSound_AsChunk(self);
+    Mix_Chunk *new_chunk = Mix_QuickLoad_RAW(old_chunk->abuf, old_chunk->alen);
+    new_chunk->volume = old_chunk->volume;
+    pgSoundObject *sound =
+        (pgSoundObject *)pgSound_Type.tp_new(Py_TYPE(self), NULL, NULL);
+    if (sound) {
+        sound->mem = NULL;
+        sound->chunk = new_chunk;
+    }
```

Comment:
I suppose you're saying the tp_new would return NULL and raise a python exception if it fails?

Is that right? That's clever.

It could be good to have a comment there explaining the if block is to try to set attributes only if not NULL, and if it's NULL the exception has already been raised.

Reply:
I actually have the if statement checker there because I kept seeing that pattern around the source code.